axe.d.ts

```diff
@@ -405,6 +408,27 @@ declare namespace axe {
     boundingClientRect: DOMRect;
   }
 
+  interface CustomNodeSerializer {
```

Comment:
I think it would be more useful to have this be a type parameter and have all functions take and return that type. It would ensure that the functions take and return the same thing.

```jsx
interface CustomNodeSerializer<T = SerialDqElement> {
  toSpec: (dqElm: DqElement) => T
  mergeSpecs(nodeSpec: T, parentFrameSpec: T) => T
}
```

Reply:
I also needed to make the nodeSerializaer.update generic to be able to take any CustomNodeSerializer (if it only takes CustomNodeSerializer<SerialDqElement> then making CustomNodeSerializer generic wouldn't really change anything.

There are still a lot of other functions that are not properly using this new type T here, like dqElmToSpec, RawNodeResult etc. Without changing the API, a lot of these functions would need to be return only generics, which is not great.

The way we use this is our code is by making our custom type extend SerialDqElement and only add optional additional properties which is not perfect but works around the type issue.